src/uct/ib/base/ib_iface.c

```diff
@@ -725,23 +725,27 @@ static int uct_ib_iface_dev_addr_is_reachable(uct_ib_iface_t *iface,
 int uct_ib_iface_is_reachable_v2(const uct_iface_h tl_iface,
                                  const uct_iface_is_reachable_params_t *params)
 {
-    uct_ib_iface_t *iface           = ucs_derived_of(tl_iface, uct_ib_iface_t);
-    const uct_ib_address_t *ib_addr = (const void*)params->device_addr;
+    uct_ib_iface_t *iface = ucs_derived_of(tl_iface, uct_ib_iface_t);
+    const uct_device_addr_t *device_addr;
     uct_iface_reachability_scope_t scope;
 
     if (!uct_iface_is_reachable_params_valid(
                 params, UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR)) {
         return 0;
     }
 
-    if (!uct_ib_iface_dev_addr_is_reachable(iface, params->device_addr)) {
+    device_addr = (const uct_device_addr_t*)
```

Comment:
do we really need casting here?
or maybe change device_addr to be const uct_ib_address_t*, update uct_ib_iface_dev_addr_is_reachable function param type, and remove casing in line 748

Reply:
fixed